web/js/blockworld.js

```diff
@@ -351,8 +356,17 @@ function getWorkspaceXMLText()
 
 function containsUnsavedChanges()
 {
+  var blank = '<xml xmlns="http://www.w3.org/1999/xhtml"></xml>';
+  // Initialize if empty
   if(!lastXML)
-    return false;
+  {
+    lastXML = getWorkspaceXMLText();
```

Comment:
This seems wrong.  You are changing lastXML without actually saving anything.  Don't you mean this:

    var blank = '<xml xmlns="http://www.w3.org/1999/xhtml"></xml>';
    return getWorkspaceXMLText() != (lastXML || blank);